src/path-util.typ

```diff
@@ -106,14 +106,18 @@
 /// Get position on path segment
 ///
 /// - s (segment): Path segment
-/// - t (float): Position (from 0 to 1)
+/// - t (float,ratio): Absolute (float) or relative (ratio) position
+///
 /// -> vector: Position on segment
-#let point-on-segment(s, t) = {
-  let (type, ..pts) = s
-  if type == "line" {
-    return point-on-polyline(s, t)
-  } else if type == "cubic" {
-    let len = bezier.cubic-arclen(..pts) * calc.min(calc.max(0, t), 1)
+#let _point-on-segment(s, t) = {
```

Comment:
Check that `t` is between 0% and 100% or 0 and the length of the segment.

Reply:
Yea thats what I was thinking. At least make note of its (and others) behaviour on this. tbh I don't think we really need to prioritise performance this much at the moment.